apps/src/code-studio/pd/foorm/FoormLibraryEditorManager.jsx

```diff
@@ -106,36 +143,56 @@ class FoormEditorManager extends React.Component {
       });
   }
 
-  initializeEmptyCodeMirror = () => {
-    this.props.setLastSaved(null);
-    this.props.setSaveError(null);
-    this.updateFormData({
-      questions: {},
-      published: null,
-      formName: null,
-      formVersion: null,
-      formId: null
+  initializeNewLibrary = () => {
+    this.props.resetAvailableLibraryQuestions([]);
+    this.props.setLibraryData({
+      name: null,
+      version: null,
+      id: null
+    });
+    this.initializeLibraryQuestion(true);
+  };
+
+  initializeLibraryQuestion = showCodeMirror => {
+    this.updateLibraryQuestionData({
+      question: {},
+      name: null,
+      id: null
     });
+
+    this.resetSaveStatus();
     this.setState({
-      showCodeMirror: true,
-      hasLoadError: false
+      hasLoadError: false,
+      showCodeMirror: showCodeMirror
     });
   };
 
-  updateFormData = formData => {
-    this.props.setFormData(formData);
+  resetSaveStatus = () => {
```

Comment:
could you update all these class property arrow functions to be bound functions instead? [this article](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1) makes a very convincing argument as to why that should be our standard practice for any function needing access to `this`.